File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int reverseNum = 0;
        int num = x;

        while(num) {
            if(reverseNum > INT_MAX/10 || reverseNum < INT_MIN/10) return 0;
            reverseNum = (reverseNum*10) + (num%10);
            num/=10;
        }

        return reverseNum;
    }
};

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        int num = x;

        while(num != 0) {
            int lastDigit = num%10;
            if(res < Integer.MIN_VALUE/10 || res > Integer.MAX_VALUE/10) {
                return 0;
            }
            res = res * 10 + lastDigit;
            num/=10;
        } 

        return res;
    }
}